engines/scumm/he/basketball/collision/bball_collision.cpp

```diff
@@ -0,0 +1,568 @@
+	_vm->_basketball->_court->_objectTree.~CCollisionObjectTree();
+	_vm->_basketball->_court->_objectList.clear();
+	_vm->_basketball->_court->_homePlayerList.clear();
+	_vm->_basketball->_court->_awayPlayerList.clear();
+
+	// Initialize the shot spots
+	_vm->_basketball->_court->_shotSpot[LEFT_BASKET].center.x = BASKET_X;
+	_vm->_basketball->_court->_shotSpot[LEFT_BASKET].center.y = BASKET_Y;
+	_vm->_basketball->_court->_shotSpot[LEFT_BASKET].center.z = BASKET_Z;
+	_vm->_basketball->_court->_shotSpot[LEFT_BASKET].radius = SHOT_SPOT_RADIUS;
+
+	_vm->_basketball->_court->_shotSpot[RIGHT_BASKET].center.x = MAX_WORLD_X - BASKET_X;
+	_vm->_basketball->_court->_shotSpot[RIGHT_BASKET].center.y = BASKET_Y;
+	_vm->_basketball->_court->_shotSpot[RIGHT_BASKET].center.z = BASKET_Z;
+	_vm->_basketball->_court->_shotSpot[RIGHT_BASKET].radius = SHOT_SPOT_RADIUS;
+
+	// Get the name and object file for this court.
+	_vm->_basketball->_court->_name = courtNames[courtID];
+
+	// Put together to relative path and filename.
+	Common::Path objectFileName = Common::Path(Common::String::format("data/courts/%s.cof", courtNames[courtID].c_str()));
+
+	// Create a file stream to the collision object file
+	Common::File objectFile;
+	if (!objectFile.open(objectFileName))
+		error("LogicHEBasketball::u32_userInitCourt(): Could not open file '%s'", objectFileName.toString(Common::Path::kNativeSeparator).c_str());
+
+	// Read in the object file version
+	char fileVersion[32];
+	int versionStringLength = objectFile.readUint32LE();
+
+	if (versionStringLength <= 0 && versionStringLength > ARRAYSIZE(fileVersion) - 1)
+		error("LogicHEBasketball::u32_userInitCourt(): Read from stream did not read the version string length correctly.");
+	
+	objectFile.read(fileVersion, versionStringLength);
+	fileVersion[versionStringLength] = '\0';
+
+	if (strcmp(fileVersion, "01.05"))
+		error("LogicHEBasketball::u32_userInitCourt(): Invalid court version field: %s", fileVersion);
+
+	// Read in the total number of objects
+	_vm->_basketball->_court->_objectCount = objectFile.readUint32LE();
+	_vm->_basketball->_court->_objectList.resize(_vm->_basketball->_court->_objectCount);
+
+	// Keep a list of pointers to the court objects...
+	CCollisionObjectVector objectPtrList;
+	objectPtrList.resize(_vm->_basketball->_court->_objectCount);
+
+	// Read in each court object...
+	for (int i = 0; i < _vm->_basketball->_court->_objectCount; i++) {
+		CCollisionBox *currentObject = &_vm->_basketball->_court->_objectList[i];
+
+		// Read in this object's description...
+		int descriptionStringLength = objectFile.readUint32LE();
+		char *tmp = (char *)malloc(descriptionStringLength + 1);
+
+		assert(tmp);
+
+		objectFile.read(tmp, descriptionStringLength);
+		tmp[descriptionStringLength] = '\0';
+
+		Common::String tmp2(tmp);
+
+		_vm->_basketball->_court->_objectList[i]._description = tmp2;
+
+		free(tmp);
+
+		// Read in all other object attributes...
```

Comment:
Check the multi-value stream reading code, added with this PR:
https://github.com/scummvm/scummvm/pull/5638

Perhaps it'll make the code here more compact

Reply:
I'd rather have it as explicit and easily debuggable as possible, for any side-by-side debugging with the original, if that's not an issue. I would prefer it less compact but so that it's easier to track down either mistakes or bad data files.

Reply:
Sure, OK :)